website/docs/r/private_key.html.md

```diff
@@ -24,43 +23,71 @@ state and does not create any external managed resources.
 ## Example Usage
 
 ```hcl
+// ECDSA key with P384 elliptic curve
 resource "tls_private_key" "example" {
   algorithm   = "ECDSA"
   ecdsa_curve = "P384"
 }
+
+// RSA key of size 4096 bits
+resource "tls_private_key" "example" {
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+// ED25519 key
+resource "tls_private_key" "example" {
+  algorithm = "ED25519"
+}
 ```
 
 ## Argument Reference
 
 The following arguments are supported:
 
-* `algorithm` - (Required) The name of the algorithm to use for
-the key. Currently-supported values are "RSA" and "ECDSA".
+* `algorithm` - (Required) The name of the algorithm to use when generating
+the key. Currently-supported values are `RSA`, `ECDSA` and `ED25519`.
+
+* `rsa_bits` - (Optional) When `algorithm` is `RSA`, the size of the generated
+RSA key in bits. Defaults to `2048`.
 
-* `rsa_bits` - (Optional) When `algorithm` is "RSA", the size of the generated
-RSA key in bits. Defaults to 2048.
+* `ecdsa_curve` - (Optional) When `algorithm` is `ECDSA`, the name of the elliptic
+curve to use. Currently-supported values are `P224` (default), `P256`, `P384` or `P521`.
 
-* `ecdsa_curve` - (Optional) When `algorithm` is "ECDSA", the name of the elliptic
-curve to use. May be any one of "P224", "P256", "P384" or "P521", with "P224" as the
-default.
+~> **Limitations when using `ECDSA` with `P224` curve**
+When using `ECDSA` with `P224`, `.public_key_openssh`, `.private_key_openssh`, `.public_key_fingerprint_md5` and `.public_key_fingerprint_sha256` will have a value of `""`.
```

Comment:
Ah, you might want to mention to see attribute reference below (due to inverse ordering). I wasn't sure why they'd have an empty fingerprint given that the PKIX key would be generated, but this is a _SSH_ fingerprint, not a [SKID/x.509 fingerprint](https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.2) (which is probably not worthwhile to generate).